src/Api/Middleware/EventLogStorageCommitMiddleware.cs

```diff
@@ -28,7 +29,11 @@ public async Task InvokeAsync(HttpContext context, IServiceProvider provider)
         {
             try
             {
-                await provider.GetRequiredService<IEventLogger>().FlushAsync();
+                var eventLoggerContext = provider.GetRequiredService<IEventLogContext>();
+                if (!string.IsNullOrEmpty(eventLoggerContext.TenantId))
```

Comment:
This breaks all event logging from an unauthenticated standpoint. I wrote some tests to validate those instances and can add them to this branch.